extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -278,4 +326,162 @@ public void transform(TransformationContext ctx) {
+            notFoundPageDisplayableEndpointProducer
+                    .produce(new NotFoundPageDisplayableEndpointBuildItem(health.rootPathUi + "/"));
+        } else if (health.alwaysIncludeUi) {
+            AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+            //we are including in a production artifact
+            //just stick the files in the generated output
+            //we could do this for dev mode as well but then we need to extract them every time
+            for (Path p : artifact.getPaths()) {
+                File artifactFile = p.toFile();
+                try (JarFile jarFile = new JarFile(artifactFile)) {
+                    Enumeration<JarEntry> entries = jarFile.entries();
+
+                    while (entries.hasMoreElements()) {
+                        JarEntry entry = entries.nextElement();
+                        if (entry.getName().startsWith(HEALTH_UI_WEBJAR_PREFIX) && !entry.isDirectory()) {
+                            try (InputStream inputStream = jarFile.getInputStream(entry)) {
+                                String filename = entry.getName().replace(HEALTH_UI_WEBJAR_PREFIX + "/", "");
+                                byte[] content = FileUtil.readFileContents(inputStream);
+                                if (IGNORE_LIST.contains(filename)) {
+                                    ClassLoader classLoader = SmallRyeHealthProcessor.class.getClassLoader();
+                                    try (InputStream resourceAsStream = classLoader
+                                            .getResourceAsStream(OWN_MEDIA_FOLDER + filename)) {
+                                        content = IoUtil.readBytes(resourceAsStream);
+                                    }
+                                }
+
+                                String fileName = HEALTH_UI_FINAL_DESTINATION + "/" + filename;
+
+                                generatedResourceProducer
+                                        .produce(new GeneratedResourceBuildItem(fileName, content));
+
+                                nativeImageResourceProducer
+                                        .produce(new NativeImageResourceBuildItem(fileName));
+
+                            }
+                        }
+                    }
+                }
+            }
+
+            Handler<RoutingContext> handler = recorder
+                    .uiHandler(HEALTH_UI_FINAL_DESTINATION, httpRootPath.adjustPath(health.rootPathUi));
+            routeProducer.produce(new RouteBuildItem(health.rootPathUi, handler));
+            routeProducer.produce(new RouteBuildItem(health.rootPathUi + "/*", handler));
+        }
+    }
+
+    private AppArtifact getHealthUiArtifact(CurateOutcomeBuildItem curateOutcomeBuildItem) {
+        for (AppDependency dep : curateOutcomeBuildItem.getEffectiveModel().getFullDeploymentDeps()) {
+            if (dep.getArtifact().getArtifactId().equals(HEALTH_UI_WEBJAR_ARTIFACT_ID)
```

Comment:
This can be called during dev mode so maybe even taking `dep.getArtifact()` into a variable may save some time? :) Not sure what is more expensive but there can be a lot of deps potentially so worth thinking about.

Reply:
Dev mode extract this to a local "cache" folder, so it's not being called on every reload. Only when the '/health' path change